common/src/main/java/org/apache/uniffle/common/segment/LocalOrderSegmentSplitter.java

```diff
@@ -105,7 +107,11 @@ public List<ShuffleDataSegment> split(ShuffleIndexResult shuffleIndexResult) {
 
         // If ShuffleServer is flushing the file at this time, the length in the index file record may be greater
         // than the length in the actual data file, and it needs to be returned at this time to avoid EOFException
-        if (dataFileLen != -1 && totalLen >= dataFileLen) {
+        if (dataFileLen != -1 && totalLen > dataFileLen) {
```

Comment:
Do we remove the `=`?

Reply:
Yes. we need. This has been discussed in #320